src/App.tsx

```diff
@@ -15,36 +17,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+interface SortingParams {
+  sortField: string;
+  isReversed: boolean;
+}
+
+enum SortType {
+  Alphabet = 'Alphabet',
+  Length = 'Length',
+  Default = '',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, isReversed }: SortingParams,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.Alphabet:
+          return good1.localeCompare(good2);
+
+        case SortType.Length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    isReversed,
+  });
+
+  const resetFilters = () => {
```

Comment:
[CODE STYLE] - Function's name should start with a verb to easily tell what your method actually does. Consider renaming `resetFilters` to something like `handleResetFilters`.